SU-HSDO/suhumsci
# Feat 274  wysiwyg utility

# READY FOR REVIEW

## Summary
- adds new table-borderless style to humsci_basic
- adds new, generic classes to humsci_basic and su_humsci_theme (not specific to decanter)
-  also adds classes to overrides of decanter areas in su_humsci_theme
- lastly, includes the config export for the wysiwyg editor changes
- there was an issue where stanford basic was the source (pointing to decanter mixins: https://github.com/SU-SWS/decanter/blob/3.0.0/scss/utilities/mixins/links/_external-link.scss) of the icon provided for the decanter-external-link style. I duplicated the image files in the su_humsci_theme and used them to point to the new "hs-external-link" class 

**TODO**
- ✅ test new classes on blocks (See screenshot here: https://github.com/SU-HSDO/suhumsci/pull/540#issuecomment-606201404)
- ✅  commit compiled su_humsci_theme CSS

## Steps to Test
1. Create a new basic page and a textarea for the **default** site. After clicking "Source" in the textarea, add the following HTML:

```html
<p class="hs-font-lead">Lead font</p>

<p class="hs-font-splash">Splash font</p>

<p class="hs-short-line-length">Short line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line lengthShort line length</p>

<p class="hs-caption">Caption caption</p>

<p class="hs-credits">Credits credits ha</p>

<p class="hs-credits"><span class="emphasized-text">Emphasized text!!!</span></p>

<p>Normal text normal normal normal sentence. Yes indeed longer <a data-entity-substitution="canonical" data-entity-type="node" data-entity-uuid="df0ba4fb-b553-410b-9533-273eb657d1b3" href="/node/491">inline link</a> sentence to test with that should be abotu right. Maybe one more sentence.</p>

<h2>Heading 2</h2>

<h3>Heading 3</h3>

<h4>Heading 4</h4>

<h5>Heading 5</h5>

<h6>Heading 6</h6>

<p><a class="hs-external-link" data-entity-substitution="canonical" data-entity-type="node" data-entity-uuid="24cc3167-1fb4-4da2-980c-fbf834b97eff" href="/node/6">Link text</a></p>

<p><a class="hs-more-link" href="google.com">Another more link</a></p>

<p><a class="hs-button" href="google.com">Another&nbsp; primary button</a></p>

<p><a class="hs-button--big" href="google.com">Another big primary&nbsp;button</a></p>

<p><a class="hs-secondary-button" href="google.com">Another&nbsp;button</a></p>

<p>&nbsp;</p>

<table>
	<thead>
		<tr>
			<th scope="col">Header 1</th>
			<th scope="col">Header 2</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>hello</td>
			<td>hello</td>
		</tr>
		<tr>
			<td>hello</td>
			<td>hlelooe</td>
		</tr>
	</tbody>
</table>

<p>&nbsp;</p>

<p>&nbsp;</p>

<table class="hs-table--borderless">
	<thead>
		<tr>
			<th scope="col">Header 1</th>
			<th scope="col">Header 2</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>hello</td>
			<td>hello</td>
		</tr>
		<tr>
			<td>hello</td>
			<td>hlelooe</td>
		</tr>
	</tbody>
</table>

<p>&nbsp;</p>

```
4. On the actual page, play around with switching the classes back and forth between the hs and decanter equivalents to see if there are any regressions.
4.  Import the config changes to the economics site using `lando drush @economics.local  config-import`.
5. On the econ site, make the same type of Basic page with the HTML from above.
6. Ensure that the econ page looks like these screenshots:

<img width="907" alt="Screen Shot 2020-03-27 at 12 54 43 PM" src="https://user-images.githubusercontent.com/12848168/77940457-ba991480-7286-11ea-9dfd-72573f7ebcdc.png">
<img width="902" alt="Screen Shot 2020-03-27 at 12 54 49 PM" src="https://user-images.githubusercontent.com/12848168/77940460-bb31ab00-7286-11ea-9cd9-67fa20844456.png">
<img width="555" alt="Screen Shot 2020-03-27 at 12 55 03 PM" src="https://user-images.githubusercontent.com/12848168/77940461-bb31ab00-7286-11ea-87d5-39bad1d28bf1.png">
<img width="556" alt="Screen Shot 2020-03-27 at 12 55 09 PM" src="https://user-images.githubusercontent.com/12848168/77940462-bb31ab00-7286-11ea-9e12-0fd56c6945d5.png">

7. Lastly, using the person detail page, or event detail page, add some of these classes to the fields and see what works:
```
a.hs-button|Primary Button
a.hs-button--big|Big Primary Button
a.hs-secondary-button|Secondary Button
a.hs-more-link|More Link
a.hs-external-link|External Link
p.hs-font-lead|Lead Font
p.hs-font-splash|Splash Font
p.hs-short-line-length|Short Line Length
p.hs-caption|Caption
p.hs-credits|Credits
span.hs-emphasized-text|Emphasized Text
table.hs-table--borderless|Borderless Table
```

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
